Fix workspaces being overwritten in individual focussing mode ISIS powder scripts

RichardWaiteSTFC opened the pull request:
**Description of work**

Now overwrite run_details.output_run_string in individual mode which as two affects: 

- Individual mode focussing produces a grouped workspace per run (as opposed to the summed mode) 
- Workspaces in the group now have only that run number in the name (previously workspaces were overwritten when the next one was added to the group as they are all given the same name including all runs).


**Report to:** Ron Smith, Paul Henry and Gabriel Perez (POLARIS)

**To test:**

(1) Follow instructions on original issue - it should now produce these workspaces

![image](https://github.com/mantidproject/mantid/assets/55979119/bb9104a9-cfa5-4320-8c49-fb8a13425c65)

(2) Check it produces the following output (on `.gsas` fil and` .nxs` per run)
![image](https://github.com/mantidproject/mantid/assets/55979119/8a961d4e-aafe-47fa-92fb-b7046036d446)

(3) Delete the `20_2` folder 

(4) Change to use `input_mode='Summed'` - it should now produce the following workspaces

![image](https://github.com/mantidproject/mantid/assets/55979119/dfea7f0b-1f30-4833-ae18-2f5cbe3ceeb2)

(5) Check it produces the following output

![image](https://github.com/mantidproject/mantid/assets/55979119/2ed25504-d4fc-499d-815c-36e5101de542)

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

Fixes #35926

<!-- delete this if you added release notes
*This does not require release notes* because **fill in an explanation of why**
If you add release notes please save them as a separate file using the Issue or PR number as the file name. Check the file is located in the correct directory for your note(s).
-->

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

RichardWaiteSTFC committed:
Overwrite run_details.output_run_string in individual mode

This has two affects: now individual mode focussing produces a grouped workspace per run (as opposed to the summed mode) and the workspaces in the group now have only that run number in the name (previously workspaces were overwritten when the next one was added to the group as they are all given the same name including all runs).

sf1919 commented:
Please ignore the osx failure. I'm assuming you will have further commits so the tests will run again at that point.

robertapplin commented:
Did you want this in the Release 6.8 milestone @RichardWaiteSTFC ?

robertapplin commented:
Code looks good. Does this need a release note, and would it be possible to add a test?

RichardWaiteSTFC committed:
Unit test for run_details.output_run_string in _output_focused_runs

This howevr does not check the output workspace naming (which occurs in common_output.split_into_tof_d_spacing_groups).
Ideally this needs some system test coverage to cehck the number and contents of workspaces in Individual batching mode - I will make a maintenence issue for this

RichardWaiteSTFC commented:
Thanks @robertapplin  - just pushed unit tests now.  Release notes to follow shortly!
As discussed offline at the stand-up I will make a maintenance issue for adding a system test for individual focussing mode

RichardWaiteSTFC committed:
Add release note

RichardWaiteSTFC commented:
Maintenance issue for adding system test created #36018 - I put milestone 6.9 for it, hope that's OK?

RichardWaiteSTFC committed:
Use helper function for unit tests

robertapplin reviewed (approved):
Thanks for adding tests and a release note :)
